infoschema/builder.go

```diff
@@ -146,11 +146,13 @@ func (b *Builder) ApplyDiff(m *meta.Meta, diff *model.SchemaDiff) ([]int64, erro
 				// While session 1 performs the DML operation associated with partition 1,
 				// the TRUNCATE operation of session 2 on partition 2 does not cause the operation of session 1 to fail.
 				tblIDs = append(tblIDs, opt.OldTableID)
-
 				b.applyPlacementDelete(placement.GroupID(opt.OldTableID))
-				err := b.applyPlacementUpdate(placement.GroupID(opt.TableID))
-				if err != nil {
-					return nil, errors.Trace(err)
+				if opt.TableID != -1 {
+					// (!= -1) => fetching new placement rules from PD
+					err := b.applyPlacementUpdate(placement.GroupID(opt.TableID))
+					if err != nil {
+						return nil, errors.Trace(err)
```

Comment:
you can put the test case of the issue into the unit test.

Reply:
maybe we can add it in the automated test ( be tested in every CI round)

Reply:
> seems applyDiff logic can't be tested in the unit test

We can, but need a more cleaner API and in-process TiDB. I mentioned that in my WIP sharing.

> maybe we can add it in the automated test

Yep, automated-test can handle it. This problem, or the original issue is not closely related to placement rules. It seems that we did not test DDL/transaction things in CI. Do we run those tests before 5.0-rc? 5.0-rc should have the bug already.

Reply:
`seems that we did not test DDL/transaction things in CI`, yes, multi-node sync logic for DDL hasn't been tested. I think we should, how do you think @bb7133 @djshow832